crates/ibc/src/core/ics03_connection/version.rs

```diff
@@ -32,9 +32,33 @@ pub struct Version {
 }
 
 impl Version {
+    /// Check whether or not the given version is compatible with supported versions
+    pub fn verify_version_supported(
+        &self,
+        supported_versions: &[Version],
+    ) -> Result<(), ConnectionError> {
+        let maybe_version_supported = supported_versions
+            .iter()
+            .find(|sv| sv.identifier == self.identifier)
+            .ok_or(ConnectionError::VersionNotSupported {
+                version: self.clone(),
+            })?;
+
+        for feature in self.features.iter() {
+            maybe_version_supported.verify_feature_supported(feature.to_string())?;
+        }
+        Ok(())
+    }
+
     /// Checks whether or not the given feature is supported in this version
-    pub fn is_supported_feature(&self, feature: String) -> bool {
-        self.features.contains(&feature)
+    pub fn verify_feature_supported(&self, feature: String) -> Result<(), ConnectionError> {
+        if feature.trim().is_empty() {
```

Comment:
this check is redundant, since for an empty feature, the next check will fail (i.e. `self.features` does not contain `""`)

Reply:
[fdab426](https://github.com/cosmos/ibc-rs/pull/626/commits/fdab426831e6f160d5646ad19c1349ed26b4021b)